eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  I. Non-static Logger instance
+ *  II. Logger variable name should be 'logger'
+ *  III. Guard with conditional block check whenever logger's logging method called.
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String LOGGER_CLASS = "Logger";
```

Comment:
Maybe using reflection to get the name of the Logger class and package name rather than a hardcoded string.